lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -1747,109 +1733,137 @@ bool SwiftLanguage::GetFunctionDisplayName(
           sc.function->GetBlock(true).GetBlockVariableList(true);
     }
 
-    if (inline_info) {
-      s << display_name;
-      s.PutCString(" [inlined] ");
-      display_name = inline_info->GetName().GetString();
-    }
-
     VariableList args;
     if (variable_list_sp)
       variable_list_sp->AppendVariablesWithScope(eValueTypeVariableArgument,
                                                  args);
-    if (args.GetSize() == 0) {
-      s << display_name;
-      return true;
+
+    s << display_name;
+    if (inline_info) {
+      s.PutCString(" [inlined] ");
     }
-    const char *cstr = display_name.data();
-    const char *open_paren = strchr(cstr, '(');
-    const char *close_paren = nullptr;
-    const char *generic = strchr(cstr, '<');
-    // If before the arguments list begins there is a template sign
-    // then scan to the end of the generic args before you try to find
-    // the arguments list.
-    if (generic && open_paren && generic < open_paren) {
-      int generic_depth = 1;
-      ++generic;
-      for (; *generic && generic_depth > 0; generic++) {
-        if (*generic == '<')
-          generic_depth++;
-        if (*generic == '>')
-          generic_depth--;
-      }
-      if (*generic)
-        open_paren = strchr(generic, '(');
-      else
-        open_paren = nullptr;
+    s << GetFunctionDisplayArgs(sc, args, exe_ctx);
+    return true;
     }
-    if (open_paren) {
-      close_paren = strchr(open_paren, ')');
     }
+    return false;
+}
+
+std::string SwiftLanguage::GetFunctionName(const SymbolContext &sc,
+                                           const ExecutionContext *exe_ctx) {
+  if (!sc.function)
+    return "";
+  if (sc.function->GetLanguage() != eLanguageTypeSwift)
+    return "";
+  std::string name = SwiftLanguageRuntime::DemangleSymbolAsString(
+      sc.function->GetMangled().GetMangledName().GetStringRef(),
+      SwiftLanguageRuntime::eSimplified, &sc, exe_ctx);
+  if (name.empty())
+    return "";
+  const char *cstr = name.data();
+  const char *open_paren = strchr(cstr, '(');
+  const char *generic = strchr(cstr, '<');
+  if (open_paren && generic && generic < open_paren)
+    return std::string(cstr, generic);
+  if (open_paren)
+    return std::string(cstr, open_paren);
```

Comment:
Hmmm why is this new logic needed?

Reply:
We didn't previously do this in the `NoArgs` case right?

Reply:
I reverted the changes that were in the `eNoArgs` case so that the PR does not break existing users. Is that what you meant?

Reply:
Is it possible that a client is using this case incorrectly? They would not get any string from this function if that's the  case.

Reply:
> Is it possible that a client is using this case incorrectly? They would not get any string from this function if that's the case.

You're saying, if we removed the case, then a client would just not get any string if `eFunctionNameNoArgs` was used? Yea if that's the case lets just leave it alone for now